rust/opendp/src/meas/histogram_alp/mod.rs

```diff
@@ -0,0 +1,398 @@
+use std::collections::HashMap;
+use std::hash::{Hash, Hasher};
+use std::rc::Rc;
+
+use num::{Integer, ToPrimitive};
+use rug::{Float, float::Round, ops::AddAssignRound, ops::DivAssignRound};
+
+use crate::core::{Measurement, Function, PrivacyRelation};
+use crate::dist::{L1Distance, MaxDivergence};
+use crate::dom::{AllDomain, MapDomain, SizedDomain};
+use crate::error::Fallible;
+use crate::interactive::Queryable;
+use crate::traits::{DistanceConstant, CheckNull, InfCast};
+use crate::samplers::{fill_bytes, CastInternalReal, SampleBernoulli};
+use std::collections::hash_map::DefaultHasher;
+
+const ALPHA_DEFAULT : u32 = 4;
+const SIZE_FACTOR_DEFAULT : u32 = 50;
+
+// Implementation of a mechanism for representing sparse histogram.
+// The mechanism was introduced in the paper:
+// "Differentially Private Sparse Vectors with Low Error, Optimal Space, and Fast Access"
+// Available here: arxiv.org/abs/2106.10068
+
+type SizedHistogramDomain<K, C> = SizedDomain<MapDomain<AllDomain<K>, AllDomain<C>>>;
```

Comment:
Based on our DMs, I'm thinking we could drop SizedDomain here. The input data doesn't need to be of exactly size n, you just need a reasonable estimate of n to derive the size of the projection that will maximize utility (by properly balancing the odds of hash collisions against the size of the release).

This will make it fail to chain with `make_count_by`, but the SizedDomain emitted from that constructor can be dropped as well. If we do need a `count_by` with sized output domains in the future, we'd name it `make_sized_count_by` anyways.